--- awesome-3.0-orig/awesomeConfig.cmake	2008-09-18 20:15:03.000000000 +0200
+++ awesome-3.0/awesomeConfig.cmake	2008-09-18 20:16:07.000000000 +0200
@@ -18,7 +18,7 @@
 option(GENERATE_MANPAGES "generate manpages" ON)
 option(GENERATE_LUADOC "generate luadoc" ON)
 
-link_directories(/usr/local/lib)
+link_directories(/usr/lib)
 
 # {{{ CFLAGS
 add_definitions(-std=gnu99 -ggdb3 -fno-strict-aliasing -Wall -Wextra
@@ -54,7 +54,7 @@
 a_find_program(XMLTO_EXECUTABLE xmlto FALSE)
 a_find_program(GZIP_EXECUTABLE gzip FALSE)
 # lua documentation
-a_find_program(LUA_EXECUTABLE lua FALSE)
+a_find_program(LUA_EXECUTABLE lua51 FALSE)
 a_find_program(LUADOC_EXECUTABLE luadoc FALSE)
 # doxygen
 include(FindDoxygen)
