diff --git a/utils/awesome-client b/utils/awesome-client
index a125869..e99faf3 100755
--- a/utils/awesome-client
+++ b/utils/awesome-client
@@ -1,7 +1,8 @@
 #!/bin/sh
 
 # rlwrap provides readline to stuff which doesn't know readline by itself
-RLWRAP=`which rlwrap 2>/dev/null`
+RLWRAP=$(which rlwrap 2>/dev/null)
+READ_OPTIONS="-r"
 
 if [ "$RLWRAP" != "" ]
 then
@@ -10,7 +11,15 @@ then
         A_RERUN="no" exec $RLWRAP $0
     fi
 else
-    USE_READLINE="-e"
+    if [ "$BASH" ]
+    then
+        READ_PARAMETERS=" -e"
+    fi
+fi
+
+if [ $BASH ]
+then
+    READ_OPTIONS="$READ_OPTIONS -p awesome# "
 fi
 
 DBUS_SEND=dbus-send
@@ -32,7 +41,7 @@ a_dbus_send()
         $DBUS_METHOD string:"$1" | tail -n +2
 }
 
-while read ${USE_READLINE} -r -p "awesome# " line
+while read ${USE_READLINE} ${READ_OPTIONS} line
 do
     a_dbus_send "$line"
 done
